### PR TITLE
(tensors) feat: add pathlib support to Tensors

### PR DESCRIPTION
**Pull request by stefpi**
cherry picked pathlib support from https://github.com/tinygrad/tinygrad/pull/6146 and added regression test for it in tests/test_tensor_data.py.

### CODE REVIEW

File: test/test_tensor_data.py [hunk 1]
```diff
@@ -64,5 +64,12 @@ def test_data_float16(self):
     assert dat.shape == (2,2)
     # NOTE: python can't deref float16
 
+  def test_data_pathlib(self):
```
Comment (geohot) [inline test/test_tensor_data.py:67]: This is not a good test for various reasons. It also doesn't test all the functionality

Comment (stefpi) [reply test/test_tensor_data.py:67]: :+1: I'll try again. What would be the ideal spot to put it in your opinion?

Comment (geohot) [reply test/test_tensor_data.py:67]: http://www.catb.org/~esr/faqs/smart-questions.html

Comment (geohot) [reply test/test_tensor_data.py:67]: Have you read all the other tests?

Comment (stefpi) [reply test/test_tensor_data.py:67]: Apparently didn't read enough of them. I'll go find a better spot.

Comment (stefpi) [reply test/test_tensor_data.py]: Moved to /test/test_tensor.py. Also made some changes to the test to improve cleanup. Based on the README of the CIFAR dataset, I moved the test to be more focused on numpy arrays that are stored in bin files. Hope this is more suitable. Hope this is an improvement, if not, please let me know. I would like to keep trying.

**Pull request closed without merging**